internal/chooserbenchmark/context.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package chooserbenchmark
+
+import (
+	"fmt"
+	"go.uber.org/multierr"
+	"sync"
+	"time"
+)
+
+type Context struct {
```

Comment:
I'm not a huge fan of `context` type structs (similar with `dependencies`/`deps`/etc.) as it seems to create a dumping group when we want to extend this. 

Your test structure looks clean so far. I'd suggest holding off from making this change since it may increase complexity -- we can leave it as is and when we create the new PR for the team, discuss it there.

Reply:
If not use context, we will have to separate the fields (including client/server references, start, stop signals) in context and pass them among different modules within benchmark. I created this type to aggregate these trivial fields together so things won't get messed up.